specification/network/resource-manager/Microsoft.Network/stable/2018-08-01/availableDelegations.json

```diff
@@ -0,0 +1,170 @@
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. Returns all of the possible delegations for a subnet in this subscription in the region.",
+            "schema": {
+              "$ref": "#/definitions/AvailableDelegationsResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get available delegations": {
+            "$ref": "./examples/AvailableDelegationsSubscriptionGet.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/locations/{location}/availableDelegations": {
+      "get": {
+        "operationId": "AvailableResourceGroupDelegations_List",
+        "description": "Gets all of the available subnet delegations for this resource group in this region.",
+        "parameters": [
+          {
+            "name": "location",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The location of the domain name."
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. Returns all of the possible delegations for a subnet in this subscription in the region.",
+            "schema": {
+              "$ref": "#/definitions/AvailableDelegationsResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get available delegations in the resource group": {
+            "$ref": "./examples/AvailableDelegationsResourceGroupGet.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AvailableDelegationsResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AvailableDelegation"
+          },
+          "description": "An array of available delegations."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URL to get the next set of results."
+        }
+      },
+      "description": "An array of available delegations."
+    },
+    "AvailableDelegation": {
+      "properties": {
+        "name": {

```

Comment:
This will conflict with the `SubResource` properties
Also, if this is a resource, please allof on a resource instead and set the readonly properties accordingly